tests/rustdoc/typedef-inner-variants.rs

```diff
@@ -0,0 +1,114 @@
+// This test checks different combinations of structs, enums, and unions
+// for the "Show Aliased Type" feature on type definition.
+
+#![crate_name = "inner_variants"]
+
+// aux-build:cross_crate_generic_typedef.rs
+extern crate cross_crate_generic_typedef;
+
+pub struct Adt;
+pub struct Ty;
+pub struct TyCtxt;
+
+pub trait Interner {
+    type Adt;
+    type Ty;
+}
+
+impl Interner for TyCtxt {
+    type Adt = Adt;
+    type Ty = Ty;
+}
+
+// @has 'inner_variants/type.AliasTy.html'
+// @count - '//*[@id="variants"]' 0
+// @count - '//*[@id="fields"]' 0
+pub type AliasTy = Ty;
+
+// @has 'inner_variants/enum.IrTyKind.html'
+pub enum IrTyKind<A, I: Interner> {
+    /// Doc comment for AdtKind
+    AdtKind(<I as Interner>::Adt),
+    /// and another one for TyKind
+    TyKind(<I as Interner>::Adt, <I as Interner>::Ty),
```

Comment:
Strictly speaking, you don't need to fully qualify the associated type but it's fine.

```suggestion
    AdtKind(I::Adt),
    /// and another one for TyKind
    TyKind(I::Adt, I::Ty),
```

Reply:
:+1: Changed for all, except for `Ty` in case it makes any difference :smile:.